pymeasure/instruments/hp/hp3437A.py

```diff
@@ -0,0 +1,412 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import math
+import struct
+from enum import IntFlag
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+from pymeasure.instruments.hp.hphelper import HPsupport
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP3437A(Instrument):
+    """Represents the Hewlett Packard 3737A system voltmeter
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super(HP3437A, self).__init__(
+            resourceName,
+            "Hewlett-Packard HP3437A",
+            includeSCPI=False,
+            send_end=True,
+            read_termination="\r\n",
+            write_termination="\r\n",
+            **kwargs,
```

Comment:
> I think, `read_termination` is not used anywhere, since when you are reading from instrument, you use `read_raw`. Correct ? From what I understood from your code, all message to instrument requires a termination '\r\n' except binary program ('B'). Is this correct ? If this the case, it would be possible override `write` and `values` methods as follow:

You are right. `read_termination` is not needed in this case.
However when reading data in ASCII format it is used, I think. 
Will give it a closer look.

> ```
> def write(self, command):
>     """ This is needed only if 'B' cannot be followed by a termination character (to be checked) """
>     if command == "B":
>         self.adapter.connection.write("B", termination="")
>     else:
>         super().write(command)
> ```
This could be probably be omittied, as I do not plan at the moment to implement the write part of the `B`-command.


>```
> def values(self, command, **kwargs):
>     if command == "B":
>         self.write(command)
>         return self.adapter.connection.read_raw(**kwargs)
>     else:
>         return super().values(command, **kwargs)
> ```
Yeah that would be one idea, will givit ia try.

> Then a each field of the status property could be defined as follow:
> 
> ```
> <field name> = Instrument.measurement("B",
>                                 get_process = <function to decode the specific field in the status>,
>                         )
> ```
> Again, some ideas as a starting point for discussions and not a definitive request to make changes.
> 
Your feedback is **much appreciated** and I will definitely give your proposal a try after I got the basic things working with the `HPLegacyInstrument `class.